File: SpaceTrader/app/src/main/java/com/example/spacetrader/model/GameState.java
package com.example.spacetrader.model;

import java.util.Random;

public class GameState {

    public enum Difficulty {
        BEGINNER, EASY, NORMAL, HARD, IMPOSSIBLE
    }

    public Random rng;
    private Universe universe;
    private Player player;
    private final Difficulty currentDifficulty;

    private static GameState sharedGameState;

    private GameState(Difficulty difficulty) {
        this.currentDifficulty = difficulty;
        this.rng = new Random(); // maybe we'll add the ability to specify a seed later.
    }

    public static GameState generateGame(String playerName, int pilotSkill, int fighterSkill, int traderSkill, int engineerSkill, Difficulty difficulty) {
        sharedGameState = new GameState(difficulty);
        sharedGameState.universe = new Universe();
        sharedGameState.player = new Player(playerName, pilotSkill, fighterSkill, traderSkill, engineerSkill);
        return sharedGameState;
    }

    public static GameState getState() {
        return sharedGameState;
    }

    public Universe getUniverse() {
        return universe;
    }

    public Player getPlayer() {
        return player;
    }

    public Difficulty getDifficulty() {
        return currentDifficulty;
    }

}

File: SpaceTraderiOS/Space Trader/Model/GameState.swift
//
//  GameState.swift
//  Space Trader
//
//  Created by Donny Lawrence on 3/2/19.
//  Copyright © 2019 Donny Lawrence. All rights reserved.
//

import Foundation

class GameState {
  enum Difficulty: String, CaseIterable {
    case beginner = "Beginner"
    case easy = "Easy"
    case normal = "Normal"
    case hard = "Hard"
    case impossible = "Impossible"
  }
  static let MAX_POINTS: Int = 16
  
  let currentDifficulty: Difficulty
  let universe: Universe
  let player: Player
  
  init(difficulty: Difficulty) {
    currentDifficulty = difficulty
    
    universe = Universe()
    universe.generateUniverse()
    
    player = Player(currentSystem: universe.solarSystems[0])
  }
}
